src/main/java/com/spotify/futures/FuturesExtra.java

```diff
@@ -159,6 +159,62 @@ public void onFailure(Throwable t) {
   }
 
   /**
+   * Represents an operation that accepts a single input argument and returns no result.
+   */
+  public interface Consumer<T> {
+
+    /**
+     * Performs this operation on the given argument.
+     * @param t the input argument
+     */
+    void accept(T t);
```

Comment:
IMO it should not be declared to throw checked exceptions.
In this context it will be invoked in the onSuccess | onFailure methods of FutureCallback, and they don't throw (guava will trap and log if it does though...nothing else will happen)
http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/util/concurrent/FutureCallback.html